src/components/ArtistName/ArtistName.jsx

```diff
@@ -0,0 +1,34 @@
+import "./artistname.css";
+
+export const ArtistName = ({ singleArtistItem }) => {
+  const { artists } = singleArtistItem;
+
+  // change how the artist name(s) is presented based of number of artists
+  const artistName = artists.length === 1
+    ? artists[0].name // this display a single artist's name
+    : artists.length === 2
+      ? `${artists[0].name} & ${artists[1].name}` // this display two artists with " & " between them
+      : artists.slice(0, -1).map(artist => `${artist.name}, `).join('') + `& ${artists.slice(-1)[0].name}`; // this display multiple artists with commas and the last artist with " & "
+
+  // create an object to hold artist information
+  const artistObjects = {
+    artistName, // formatted artist name
+    aHref: singleArtistItem.artists[0].external_urls.spotify, // link to artist's Spotify page
```

Comment:
This is the oooonly possible problem I can see, in case of two artists, we still only get the spotify url for the first artist in the array. This was super tricky tho!